src/main/java/uk/gov/hmcts/reform/hmc/service/HearingManagementServiceImpl.java

```diff
@@ -63,26 +64,38 @@ public void validateRequestVersion(HearingDetailsRequest hearingDetailsRequest,
     }
 
     private void isValidRequest(HearingDetailsRequest hearingDetailsRequest) {
-        log.info("Validating hearing response");
+        log.debug("Validating hearing response");
         if (hearingDetailsRequest.getHearingResponse() != null && hearingDetailsRequest.getErrorDetails() != null) {
             throw new BadRequestException(INVALID_HEARING_PAYLOAD);
         }
 
         if (hearingDetailsRequest.getHearingResponse() != null
                 && hearingDetailsRequest.getHearingResponse().getHearing().getHearingVenue() != null) {
+
             final HearingVenue hearingVenue = hearingDetailsRequest.getHearingResponse().getHearing().getHearingVenue();
             if (!CollectionUtils.isEmpty(hearingVenue.getLocationReferences())) {
                 getLocationReference(hearingVenue.getLocationReferences());
             }
+
+            hearingDetailsRequest.getHearingResponse().getHearing().getHearingSessions().forEach(session -> {
```

Comment:
Couple of minor points:
- This is inside the `if` from L73 which is also checking that `hearingDetailsRequest.getHearingResponse().getHearing().getHearingVenue() != null` - since the top level is also optional, might want to make that check only cover L75-78
- `hearingSessions` can also be null so worth adding a check for this new block
-

Reply:
Should be resolved in this update